probe/overlay/weave.go

```diff
@@ -21,8 +27,25 @@ const (
 
 	// WeaveDNSHostname is the ket for the WeaveDNS hostname
 	WeaveDNSHostname = "weave_dns_hostname"
+
+	// WeaveMACAddress is the key for the mac address of the container on the
+	// weave network, to be found in container node metadata
+	WeaveMACAddress = "weave_mac_address"
 )
 
+var weavePsMatch = regexp.MustCompile(`^([0-9a-f]{12}) ((?:[0-9a-f][0-9a-f]\:){5}(?:[0-9a-f][0-9a-f]))(.*)$`)
```

Comment:
This is rather fragile (i.e. the output of `weave ps` change in the future). Have you considered replicating the behaviour from `weave ps` in using the docker API?